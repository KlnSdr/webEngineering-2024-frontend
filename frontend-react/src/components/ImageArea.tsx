import React from "react";


interface ImageSrc{
    origin: string;
}

const ImageArea: React.FC<ImageSrc> = ({origin}) => {
  return (
    <div className="ImageArea">
      <img src={origin} alt="Image"/>
    </div>
  );
}

export default ImageArea;