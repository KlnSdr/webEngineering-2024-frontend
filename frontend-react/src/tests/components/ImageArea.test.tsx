import React from "react";
import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import ImageArea from "../../components/ImageArea";

describe("ImageArea Component", () => {
    test("renders with correct imgUri", () => {
        const origin = "https://via.placeholder.com/150";

       const {asFragment} = render( <ImageArea origin={origin} />);

        const imageElement = screen.getByRole("img");
        expect(imageElement).toHaveAttribute("src", origin);
        expect(imageElement).toHaveAttribute("alt", "dish");
        expect(asFragment()).toMatchSnapshot();
    });
});
