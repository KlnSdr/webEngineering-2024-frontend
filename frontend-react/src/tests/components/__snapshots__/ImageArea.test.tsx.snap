// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageArea Component renders with correct imgUri 1`] = `
<DocumentFragment>
  <div
    class="ImageArea"
  >
    <img
      alt="dish"
      src="https://via.placeholder.com/150"
    />
  </div>
</DocumentFragment>
`;
